cmd/maya-apiserver/spc-watcher/handler.go

```diff
@@ -39,10 +39,11 @@ func (c *Controller) syncHandler(key, operation string, object interface{}) erro
 	// Check if the event is for delete and use the spc object that was pushed in the queue
 	// for utilising details from it e.g. delete cas template name for storagepool deletion.
 	if operation == deleteEvent {
+		object := object.(*apis.StoragePoolClaim)
 		if object == nil {
-			glog.Error("Nil storagepoolclaim object found for storage pool deletion")
+			return fmt.Errorf("storagepoolclaim object not found for storage pool deletion")
 		}
-		spcGot = object.(*apis.StoragePoolClaim)
+		spcGot = object
```

Comment:
Why did we have to change this typecast logic? If this needs to be fixed, please take it up in the next commit.

Reply:
I will add this as a part of next PR